src/function/FunctionRestriction.ts

```diff
@@ -1,63 +1,47 @@
-import {Restrictions} from "../config/Restrictions";
-import {ProjectElements, ProjectLinks, VocabularyElements} from "../config/Variables";
-import {RestrictionObject} from "../datatypes/RestrictionObject";
-import {parsePrefix} from "./FunctionEditVars";
-import {ConnectionObject} from "../datatypes/ConnectionObject";
+import {ProjectElements, ProjectLinks, ProjectSettings, VocabularyElements} from "../config/Variables";
+import {Restriction} from "../datatypes/Restriction";
+import {getNewLink} from "./FunctionGetVars";
+import {LinkType} from "../config/Enum";
+import {addLink} from "./FunctionCreateVars";
 
-export function createRestriction(obj: { [key: string]: any }, iri: string, restriction: string, onProperty: string, target: { type: string, value: string }) {
-	if (target.type !== "bnode" && (restriction in Restrictions)) {
-		let newRestriction = new RestrictionObject(restriction, onProperty, target.value);
+export function createRestriction(obj: { [key: string]: any }, iri: string, restriction: string, onProperty: string, target: { type: string, value: string }, onClass?: string) {
```

Comment:
Two points regarding `obj: { [key: string]: any }`:

1) There is a handy Record utility type that can be used in similar cases
https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype

2) The `any` would deserve better specification, I am sure the objects types that can come in are known so it would be great to have it replaced i.e. by "SomeType | OtherType | MaybeAThirdType" insted. Alternatively it's possible to describe the interface, e.g. `Record<string, { restrictions: Restriction[] }>`

Reply:
I eventually replaced it with just Restriction[], but I will keep Records in mind - there are other places that will need it. Thank you for the comment.